ludwig/schema/utils.py

```diff
@@ -56,6 +58,31 @@ def load_config_with_kwargs(
     }
 
 
+def convert_submodules(config_dict: dict) -> TDict[str, any]:
```

Comment:
Nit: If its only used by the config object, feel as though this belongs in the config_object module itself

Reply:
So it actually gets used in the schema utils file as well in the `BaseMarshmallowConfig` since I wanted it to be inherited by all schema classes so that any conversion to dict will convert submodules.